### PR TITLE
blocklist front end for the admin panel

### PR DESCRIPTION
**Pull request by tobiasd**
@Hypolite / @annando wanted to implement the back-end IIRC the front-end  does not contain a hint that it's only dummy function so far ;-)

### CODE REVIEW

**Comment by tobiasd** [comment]
Just saw, that #3387 is an issue tangented by this PR, so referring to it.

File: mod/admin.php [hunk 1]
```diff
@@ -253,6 +260,93 @@ function admin_content(App $a) {
...
+		'$newreason' => Array('newentry_reason', t('Block reason'), '', t('The reason why you blocked this URL.'), 'required', '', ''),
+		'$submit' => t('Add Entry'),
+		'$savechanges' => t('Save changes to the blocklist'),
+		'$currenttitle' => t('Current Entries in the Blocklist'),
+		'$thurl' => t('Blocked URL'),
+		'$threason' => t('Reason for the block'),
+		'$delentry' => t('Delete entry from blocklist'),
+		'$entries' => $blocklistform,
+		'$baseurl' => App::get_baseurl(true),
+		'$confirm_delete' => t('Delete entriy from blocklist?'),
```
Comment (Hypolite) [inline mod/admin.php]: entry*

File: view/templates/admin_blocklist.tpl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	}
+</script>
+<div id="adminpage">
+  <h1>{{$title}} - {{$page}}</h1>
+  <p>{{$intro}}</p>
+	<p>{{$public}}</p>
+
+	<h2>{{$addtitle}}</h2>
+	<form action="{{$baseurl}}/admin/blocklist" method="post">
+    <input type='hidden' name='form_security_token' value='{{$form_security_token}}'>
```
Comment (Hypolite) [inline view/templates/admin_blocklist.tpl]: Standards: Inconsistent indentation through this file.
Please use double-quotes for attributes.

File: mod/admin.php [hunk 3]
```diff
@@ -253,6 +260,93 @@ function admin_content(App $a) {
...
+ * adding, removing and editing of entries from the blocklist.
+ *
+ * @param App $a
+ * @return string
+ */
+function admin_page_blocklist(App $a) {
+	$blocklist = Config::get('system', 'blocklist');
+	$blocklistform = array();
+	if (is_array($blocklist)) {
+		foreach($blocklist as $id =>$b) {
```
Comment (Hypolite) [inline mod/admin.php]: Standards: can you please add a space after `=>`?

File: mod/admin.php [hunk 4]
```diff
@@ -253,6 +260,93 @@ function admin_content(App $a) {
...
+			$blocklistform[] = Array(
```
Comment (Hypolite) [inline mod/admin.php]: Standards: Please use lowercase `array` for native variable types. We only use CamelCase for class names.

File: view/templates/admin_blocklist.tpl [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<script>
+	function confirm_delete(uname){
+		return confirm( "{{$confirm_delete}}".format(uname));
```
Comment (Hypolite) [inline view/templates/admin_blocklist.tpl]: Standards: Please remove the space after the opening brace.

File: view/templates/admin_blocklist.tpl (hunk 2 above)
Comment (tobiasd) [inline view/templates/admin_blocklist.tpl]: I've exchanged all the `\t` now manually with `\t` ;-)

**Pull request merged**